### DISCUSSION TITLE
FYI, "zyaml" project has partial implementation of Zigbee specification in YAML

### DISCUSSION THREAD

**Discussion by Hedda** [ffenix113/zigbee_home#52] [General]
FYI, the "zyaml" project look to contain a partial implementation of Zigbee specification in YAML representation:

https://github.com/hemtjanst/zyaml

Stumbled upon that project and not sure if it actually contains any concepts or ideas that could gotten from it?

**Comment by ffenix113** [comment]
Thank you, I did see the project a while back, and while it has some useful information - it is a bit too complex and "specialized" for our use case.

Though in the future maybe we could use the generator to add the clusters.

**Comment by Hedda** [comment]
> Though in the future maybe we could use the generator to add the clusters.

Might also be able to use the Connectivity Standards Alliance's project-chip (Matter) project's "ZAP" to generate clusters and attributes?

https://github.com/project-chip/zap

ZAP (which stands for "ZCL Advanced Platform" ) is a generic generation engine and user interface for applications and libraries based on Zigbee Cluster Library, the specification developed by the [Connectivity Standards Alliance](https://csa-iot.org/) to configure Matter and Zigbee Pro applications.

**Reply by ffenix113** [reply]
Probably. It does looks nice, so I will try to use it when will have some time for it. Thank you!
